File: app/src/main/java/com/xmartlabs/template/module/ControllerModule.java
package com.xmartlabs.template.module;

import com.xmartlabs.bigbang.core.controller.CoreSessionController;
import com.xmartlabs.template.model.Session;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ControllerModule {
  @Provides
  @Singleton
  CoreSessionController provideSessionController() {
    return new CoreSessionController(Session.class);
  }
}

File: app/src/main/java/com/xmartlabs/template/module/ControllerModule.kt
package com.xmartlabs.template.module

import com.xmartlabs.bigbang.core.controller.SharedPreferencesSource
import com.xmartlabs.template.controller.SessionRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ControllerModule {
  @Provides
  @Singleton
  internal fun provideSessionController(sharedPreferencesSource: SharedPreferencesSource) = SessionRepository(sharedPreferencesSource)
}
